pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -408,7 +413,41 @@ public void testUploadSegmentRefreshOnly()
       assertEquals(segmentsZKMetadata.size(), 1);
       assertNotEquals(segmentsZKMetadata.get(0).getRefreshTime(), Long.MIN_VALUE);
     }
+    waitForNumOfSegmentsBecomeOnline(offlineTableName, 1);
     dropOfflineTable(SEGMENT_UPLOAD_TEST_TABLE);
+    testTableDataManagerCleanup(offlineTableName);
+  }
+
+  private void waitForNumOfSegmentsBecomeOnline(String tableNameWithType, int numSegments)
+      throws InterruptedException, TimeoutException {
+    long endTimeMs = System.currentTimeMillis() + EXTERNAL_VIEW_ONLINE_SEGMENTS_MAX_WAIT_MS;
+    do {
+      Set<String> onlineSegments = _helixResourceManager.getOnlineSegmentsFromExternalView(tableNameWithType);
+      if (onlineSegments.size() == numSegments) {
+        return;
+      }
+      Thread.sleep(EXTERNAL_VIEW_CHECK_INTERVAL_MS);
+    } while (System.currentTimeMillis() < endTimeMs);
+    throw new TimeoutException(String
+        .format("Time out while waiting segments become ONLINE. (tableNameWithType = %s)", tableNameWithType));
+  }
+
+  private void testTableDataManagerCleanup(String offlineTableName)
```

Comment:
Is it possible to put this method to `BaseClusterIntegrationTestSet` since both offline and realtime cluster integration tests have this method respectively?

Reply:
Good catch, refactored the function and put it in the parent class.